README.md

```diff
@@ -60,6 +63,52 @@ You should see output similar to the following:
  * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
  * Restarting with fsevents reloader
  * Debugger is active!
- * Debugger PIN: 226-556-590
+ * Debugger PIN: 999-999-999
 ```
 Now visit [`http://localhost:5000/`](http://localhost:5000/) in your web browser to view the app.
+
+## Running the App on a VM (Vagrant)
+
+This project comes with a Vagrant file that allows the ToDo app to run a VM.  You will need to download and install the following two software: Virtual Box, https://www.virtualbox.org/ and Vagrant, https://www.vagrantup.com/.  Open up a terminal console, and navigate to this project folder, type:
+```bash 
+vagrant up
+```
+If this is not done before, it will take several minutes to download an Ubuntu image from hashicorp site and do various configuration.
+
+At the end, the output should be similar in above section on Running The App.
+
+To end the Vagrant session, CTRL-C and kill the ruby process.
+
+## Running the App on Docker
+This project also comes with a file `dockerFile` that have two profiles: `developments` and `production`
+
+### Creating and running a container for development
+In a console, run this command:
+
+```powershell
+docker build --target developments --tag todo-app:dev .
+```
+To run the app server from the container:
+```powershell
+docker run -p 5000:5000 --env-file .\.env todo-app:dev
```

Comment:
The exercise instructions ask you to run it with a bind mount so that you can take advantage of reloading the app after file changes without having to rebuild and re-run the image.